docs/core/additional-tools/xmlserializergenerator-instructions.md

```diff
@@ -0,0 +1,106 @@
+---
+title: Using Microsoft XML Serializer Generator on .NET Core
+description: An overview of the Microsoft XML Serializer Generator.
+author: mlacouture
+manager: wpickett
+ms.author: johalex
+ms.date: 12/20/2017
+ms.topic: tutorial
+ms.prod: .net-core
+ms.custom: mvc
+---
+# Using Microsoft XML Serializer Generator on .NET Core
+
+This tutorial teaches you how to use the Microsoft XML Serializer Generator in a C# .NET Core application. During the course of this tutorial, you learn:
+
+> [!div class="checklist"]
+> * How to create a .NET Core app.
+> * How to add a reference to the Microsoft.XmlSerializer.Generator package
+> * How to edit your MyApp.csproj to add dependencies.
+> * How to add a class and an XmlSerializer
+> * How to build and run the application 
+
+Like the [Xml Serializer Generator (sgen.exe)](https://docs.microsoft.com/en-us/dotnet/standard/serialization/xml-serializer-generator-tool-sgen-exe) for .NET Framework, <xref:System.Xml.Serialization.XmlSerializer> NuGet package is the equivalent for .NET Core and .NET Standard projects. It creates an XML serialization assembly for types contained in an assembly to improve the startup performance of XML serialization when serializing or de-serializing objects of those types using XmlSerializer.
+
+## Prerequisites
+
+To complete this tutorial:
+
+* [.NET Core SDK 2.1.3 or later](https://www.microsoft.com/net/download)
+* Install your favorite code editor, if you haven't already.
+
+> [!TIP]
+> Need to install a code editor? Try [Visual Studio](https://visualstudio.com/downloads)!
+  
+## Use Microsoft XML Serializer Generator in a .NET Core console application 
+
+The following instructions show you how to use XML Serializer Generator in a .NET Core console application.
+
+### Create a .NET Core console application
+
+Open a command prompt and create a folder named *MyApp*. Navigate to the folder you created and type the following commands:
+
+```console
+dotnet new console --name MyApp
+```
+
+### Add a reference to the Microsoft.XmlSerializer.Generator package in the MyApp project.
+
+ Type:
+ 
+ ```console
+ dotnet add MyApp package Microsoft.XmlSerializer.Generator -v 1.0.0-preview1-25915-02
```

Comment:
When I tried this command, it created a normal `<PackageReference>`, not a `<DotNetCliToolReference>`. So, isn't it necessary to edit the csproj by hand?

See also https://github.com/NuGet/Home/issues/4901.

Reply:
Yes, `<DotNetCliToolReference>` needs to be added manually.

Reply:
It wasn't clear in the original article. Based on the original wording, I assumed that dotnet add created this as well.  Now revised. Thanks!